### ISSUE TITLE
Simulation Panic - Invalid Segment

### ISSUE THREAD

**Issue by CarstenBaumannH** [citybound/citybound#111]
Simulation Panic is occured when implementing a plan (extending an existing, already implemented road).
![invalid_segment](https://cloud.githubusercontent.com/assets/24792832/22529920/ae8f5bd4-e8d9-11e6-9cda-f6ad6a0fc811.PNG)

**Comment by aeickhoff** [comment]
Thank you for reporting, could you please explain/show in which order you created these roads and zoom in so more detail is visible? Thanks!

**Comment by CarstenBaumannH** [comment]
Zooming in is difficult, sorry already deleated the file :/.

First I created the cross. Then I expanded the lower to strokes with the bend.

Then I expanded the upper part where the gap is visable.

**Comment by aeickhoff** [comment]
Where the two bends meet, is there also an intersection? And it only crashed after you extended it at the top?

**Comment by CarstenBaumannH** [comment]
Yes, at the two bends is a intersection. But this was implemented right. The problem occured at the upper part of the cross intersection later.

**Comment by aeickhoff** [comment]
and the error occurred only on implementation, not during planning, correct?

**Comment by CarstenBaumannH** [comment]
Yes!

**Comment by Forecaster** [comment]
I just got this error. This is what I was doing:

1, 2, A and B refers to the linked image.

I had built road 1, then built road 2 in one go.
Then I deleted the stretch between A and B (the shortest way, crossing road 1)
I then tried to reconnect them, from A to B
While trying to align it so all the lanes would connect the error occurred.

http://imgur.com/a/EW8ZR

**Comment by aeickhoff** [comment]
There is probably a million ways to get an `invalid segment`! Thanks for reporting

**Comment by mateon1** [comment]
I got this crash by planning a short, straight road segment, followed by a massive circle _almost_ perfectly wrapping around to the start of the straight segment.
